video/video_samples.rb

```diff
@@ -0,0 +1,236 @@
+      puts "Thumbnail size: #{face_annotation.thumbnail.length}"
+
+      face_annotation.segments.each do |segment|
+        if segment.start_time_offset == -1 && segment.end_time_offset == -1
+          puts "Location #{segment_counter}: Entire video"
+        else
+          start_in_seconds = segment.start_time_offset / 1000000.0
+          end_in_seconds   = segment.end_time_offset / 1000000.0
+
+          puts "Location #{segment_counter}: #{start_in_seconds} through #{end_in_seconds}"
+        end
+
+        segment_counter += 1
+      end
+    end
+  end
+
+  puts "Processing video for face annotations:"
+  operation.wait_until_done!
+  # [END analyze_faces]
+end
+
+def analyze_safe_search path:
+  # [START analyze_safe_search]
+  # path = "Path to a video file on Google Cloud Storage: gs://bucket/video.mp4"
+
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features     = [:SAFE_SEARCH_DETECTION]
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video path, features do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+
+    annotation_result.safe_search_annotations.each do |safe_search_annotation|
+      time_in_seconds = safe_search_annotation.time_offset / 1000000.0
+
+      puts "Time: #{time_in_seconds}"
+      puts "\tadult:   #{safe_search_annotation.adult}"
+      puts "\tspoof:   #{safe_search_annotation.spoof}"
+      puts "\tmedical: #{safe_search_annotation.medical}"
+      puts "\tracy:    #{safe_search_annotation.racy}"
+      puts "\tviolent: #{safe_search_annotation.violent}"
+    end
+  end
+
+  puts "Processing video for face annotations:"
+  operation.wait_until_done!
+  # [END analyze_safe_search]
+end
+
+def analyze_shots path:
+  # [START analyze_shots]
+  # path = "Path to a video file on Google Cloud Storage: gs://bucket/video.mp4"
+
+  require "google/cloud/video_intelligence/v1beta1"
+
+  video_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
+  features     = [:SHOT_CHANGE_DETECTION]
+
+  # Register a callback during the method call
+  operation = video_client.annotate_video path, features do |operation|
+    raise operation.results.message? if operation.error?
+    puts "Finished processing."
+
+    # first result is retrieved because a single video was processed
+    annotation_result = operation.results.annotation_results.first
+    shot_counter      = 1
+
+    annotation_result.shot_annotations.each do |shot_annotation|
+      start_in_seconds = shot_annotation.start_time_offset / 1000000.0
+      end_in_seconds   = shot_annotation.end_time_offset / 1000000.0
+
+      puts "Scene #{shot_counter}: #{start_in_seconds} to #{end_in_seconds}"
```

Comment:
I love the variable name: shot_counter

But what do you think about this for consistency and simplicity:

```ruby
puts "#{start_in_seconds} through #{end_in_seconds}"
```

Users can increment numbers when looping on their own time 😋 

Or, if you want...

```ruby
puts "Scene: #{start_in_seconds} to #{end_in_seconds}"
```

^--- if we do this, consider whether it would be useful to update the others in a similar manner